scanner/scanner.go

```diff
@@ -117,8 +117,13 @@ func (s *Scanner) processEntry(info entryInfo, foundCert func(*ct.LogEntry), fou
 	}
 }
 
-func (s *Scanner) processMatcherEntry(matcher Matcher, info entryInfo, foundCert func(*ct.LogEntry), foundPrecert func(*ct.LogEntry)) error {
-	logEntry, err := ct.LogEntryFromLeaf(info.index, &info.entry)
+func (s *Scanner) processMatcherEntry(matcher Matcher, info entryInfo, foundCert func(*ct.RawLogEntry), foundPrecert func(*ct.RawLogEntry)) error {
+	rawLogEntry, err := ct.RawLogEntryFromLeaf(&info.entry, info.index)
+	if err != nil {
+		return fmt.Errorf("failed to build raw log entry %d: %v", info.index, err)
+	}
+	// Matcher instances need the parsed [pre-]certificate.
+	logEntry, err := rawLogEntry.ToLogEntry()
 	if s.isCertErrorFatal(err, logEntry, info.index) {
```

Comment:
Btw, `isCertErrorFatal` can be slightly simplified, because `IsFatal` checks the `nil` case.

Reply:
Ah, maybe not, because it has some extra logic with counter.